### ISSUE TITLE
Invalid rght values and arguably suboptimal tree_id values in the test suite (and therefore also in django-mptt)

### ISSUE THREAD

**Issue by matthiask** [django-mptt/django-mptt#483]
Let me say first that I feel kind of bad about what I'm going to say here. I'm having a hard time understanding the current mptt attributes maintenance code and experimented with replacing the recalculation of mptt attributes in Python with doing as much as possible in the database and only refreshing the mptt attributes from the database when necessary (using `refresh_from_db()`).

The initial reason for doing that was me trying to find a fix for the managers issue and getting fed up with the complexity in MPTTOptions, MPTTModelBase etc. One thing led to another and now I have a divergent code base which works, but I don't really want to fork django-mptt... (The work is here: https://github.com/matthiask/django-mptt/compare/master...mk/nomagic)

Anyway, I found a place where the current code yields incorrect results:

1. **rght is 6, but should be 4**:
- https://github.com/django-mptt/django-mptt/blob/e658dc0da2a23155ea46f07c66aa41fe238dc7a5/tests/myapp/tests.py#L929
- https://github.com/django-mptt/django-mptt/blob/e658dc0da2a23155ea46f07c66aa41fe238dc7a5/tests/myapp/tests.py#L1109

This doesn't happen with mk/nomagic anymore. There's certainly a less invasive fix, but I'm having a hard time following all that's going on with change tracking etc.

2. I always assumed that tree_id should contain consecutive integers starting with 1. This isn't the case with the current code:
- https://github.com/django-mptt/django-mptt/blob/e658dc0da2a23155ea46f07c66aa41fe238dc7a5/tests/myapp/tests.py#L467
- https://github.com/django-mptt/django-mptt/blob/e658dc0da2a23155ea46f07c66aa41fe238dc7a5/tests/myapp/tests.py#L2038

**Comment by craigds** [comment]
You're correct on both points, those both look like bugs. (1) is espcially bizarre. I guess the change tracking is getting an off-by-1 error somewhere.

I haven't had a chance to have a really good look at your nomagic branch, though I like the bits I have seen. I'll try and schedule some time to do a thorough review in the next week or so.

**Comment by matthiask** [comment]
Thank you, I appreciate it.
